conans/client/installer.py

```diff
@@ -507,7 +508,13 @@ def _build_package(self, node, output, keep_build, remotes):
         return pref
 
     @staticmethod
-    def _propagate_info(node, using_build_profile):
+    def _propagate_info(node, using_build_profile, fixed_package_id):
+        if fixed_package_id:
+            # if using config.full_transitive_package_id, it is necessary to recompute
+            # the node transitive information necessary to compute the package_id
+            # as it will be used by reevaluate_node() when package_revision_mode is used and
+            # PACKAGE_ID_UNKNOWN happens due to unknown revisions
+            node.package_id_transitive_reqs()
```

Comment:
Is there a reason to do this only when `full_transitive_package_id` is activated?

Reply:
`reevaluate_node` will run `_compute_package_id` that calls this same function. Why is it needed to update this information before doing those calls?

Reply:
Because ``reevaluate_node`` only runs for those nodes that has PACKAGE_ID_UNKNOWN, but the information that needs to be re-computed after being built are the one of the dependencies of the node with PACKAGE_ID_UNKNOWN.

Only ``full_transitive_package_id`` is causing the issue in this case, because if not defined, a different computation is done, based on ``conanfile.requires`` (which is updated).

I think the root cause of this is the immutability of the reference objects, that produces that there are several copies of ConanFileReference and PackageReference. I think for the future we should aim for a central definition of these for each node, in a way that an update to them automatically is seen by everyone else, because they contain a reference to this object.

Reply:
> Because `reevaluate_node` only runs for those nodes that has PACKAGE_ID_UNKNOWN, but the information that needs to be re-computed after being built are the one of the dependencies of the node with PACKAGE_ID_UNKNOWN.

Here you mean, the CONSUMERS of the node with `PACKAGE_ID_UNKNOWN`, right?

----

> Only full_transitive_package_id is causing the issue in this case, because if not defined, a different computation is done, based on conanfile.requires (which is updated).

Having a quick look to `conanfile.requires`, it is built using the _direct_ and the _indirect_ prefs too, so maybe we can simplify all of this a lot moving logic into ConanInfo (not this PR).

Reply:
PkgA->PkgB->PkgC

PkgA has PACKAGE_ID_UNKNOWN, because we still need to build PkgC or PkgB and awe are in ``package_revision_mode``

``reevaluate_node`` will only be executed for ``PkgA``. But the computation of transitive updated ``PackageReferences`` need to be computed for both ``PkgB`` and ``PkgC``.

Does this clarify the issue a bit?

Reply:
Ok, we are running them to populate `Node::id_direct_prefs` and `Node::id_indirect_prefs` with the new values (are we doing this twice for `pkgC`?)

Reply:
Yes, it is being called twice. That is the key of the issue, we need to run this 2 times:
- One to compute the first package_id, while analyzing the graph binaries
- A second one, while installing the binaries, to update for possible changes upstream that will be used in reevaluate_node to compute the package_id of consumers with PACKAGE_ID_UNKNOWN.

It seems it would be very challenging to skip running this for leaf nodes that wouldn't need to run this.